ml-proto/runtests.py

```diff
@@ -32,29 +37,48 @@ def _compareLog(self, fileName, logPath):
     except IOError:
       pass
 
+  def _readHeaders(self, fileName):
+    expected_exit_code = 0
+
+    with file(fileName) as f:
+      headers = f.readline().split(' ')
+
+      if headers[0] == ';;;':
+        for h in headers[1:]:
+          if h.strip() == 'expected-fail':
```

Comment:
Instead of instrumenting comments in the file, what do you think about simply using a file name convention? Say, `<test>.fail.wast` for tests that ought to fail? That's easier and cheaper to process (including for other test harnesses).